lib/plugins/package/package.js

```diff
@@ -66,6 +67,9 @@ class Package {
       'package:createDeploymentArtifacts': () => BbPromise.bind(this)
         .then(this.packageService),
 
+      'before:package:compileFunctions': () => BbPromise.bind(this)
+        .then(this.deleteNullEnvVars),
```

Comment:
This seemed like the best place to hook in to things and delete the null env vars.  That said, I could see the logic in doing it at the very beginning.